### PR TITLE
Fix for Microcks template Keycloak realm file to support Microcks 1.0.0 onwards

### PR DESCRIPTION
**Pull request by narioinc**
@EricWittmann 

I am creating this pull request for the fix to the Microcks realm template file to as to fix the bug #1319. The Fix proposes that with Microcks 1.0.0, role definitions have moved from the realm scope to the microcks-app client scope so that it is easier to plug Microcks on an existing realm.

**PR tested on the following environment:**
Environment - Ubuntu 20.04
APICurio release - BETA 2.465
Affected artefact - Microcks realm config JSON file
Potential fix available - Yes
Fix tested on - Ubuntu 20.04

@lbroudoux would like to review this PR along with you to ensure correctness. Thanks

### CODE REVIEW

**Comment by EricWittmann** [comment]
@lbroudoux Does this look good to you?

File: distro/docker-compose/config/keycloak/microcks-realm.json.template [hunk 1]
```diff
@@ -297,8 +295,28 @@
...
       "security-admin-console": [],
+      "microcks-app": [
+        {
+          "id": "b70db266-0a25-48b8-a53f-5666febd06ea",
+          "name": "user",
+          "composite": false,
+          "clientRole": true,
+          "containerId": "61397ec1-d0fa-4b9b-a38a-c301326b7833",
+          "attributes": {}
+        }
```
Comment (lbroudoux) [inline distro/docker-compose/config/keycloak/microcks-realm.json.template:318]: To be fully aligned, we should also add the `manager` and `admin` roles as in https://github.com/microcks/microcks/blob/master/install/docker-compose/keycloak-realm/microcks-realm-sample.json#L65

File: distro/docker-compose/config/keycloak/microcks-realm.json.template [hunk 2]
```diff
@@ -845,6 +875,48 @@
...
+            "multivalued": "true",
+            "user.attribute": "foo",
+            "id.token.claim": "true",
+            "access.token.claim": "true",
+            "claim.name": "groups",
+            "jsonType.label": "String"
+          }
+        }
+      ]
+    },
```
Comment (lbroudoux) [inline distro/docker-compose/config/keycloak/microcks-realm.json.template:919]: This is not related to a Microcks requirement

**Review by lbroudoux — CHANGES_REQUESTED** [review]
The `user` roles is defined twice and the `admin` and `manager` roles are missing. Also I don't see where the `user` role is defined as a default for the `microcks-app` client.

As this template should be the result of a Keycloak export, it is quite hard to understand and diff because of all the ids and optional attributes. Wouldn't it be possible to use a simpler form like in https://github.com/microcks/microcks/blob/master/install/docker-compose/keycloak-realm/microcks-realm-sample.json ?  I'll try re-used an older Keycloak version to test things up becase we've switcher to v10.1 on the Microcks side.

**Comment by narioinc** [comment]
@lbroudoux  @EricWittmann  I guess the template file can be changed to the version that is there in microcks github.
Since I saw that the current template file seems to have been created from an export with all the finer details, I too took did a export and sent in a PR. 
I will create a PR with just the realm config provided mi microcks team here
https://github.com/microcks/microcks/blob/master/install/docker-compose/keycloak-realm/microcks-realm-sample.json

**Comment by narioinc** [comment]
I have changed the file to reflect the correct concise template needed for the Microcks app to run. Checked a quick install it works on my environment

**Review by EricWittmann — APPROVED** [review]
LGTM waiting for @lbroudoux to approve.

**Review by lbroudoux — APPROVED** [review]

**Comment by lbroudoux** [comment]
LGTM too ! Thanks @narioinc for PR !

**Comment by narioinc** [comment]
thanks @lbroudoux for the review and approval !

**Pull request merged**